src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1092,10 +1092,11 @@ void CodeGen::genHWIntrinsic_R_R_RM_I(
 
     if (ins == INS_insertps)
     {
-        if (op1Reg == REG_NA)
+        if (op1->isContained())
         {
             // insertps is special and can contain op1 when it is zero
-            assert(op1->isContained() && op1->IsVectorZero());
+            assert(op1->IsVectorZero());
+            assert(op1Reg == REG_NA);
```

Comment:
We don't want this assert because registers can sometimes be set even if they're unused, such as to support spilling or use/defs